helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -683,6 +694,27 @@ public void onCustomizedViewChange(List<CustomizedView> customizedViewList,
     }
   }
 
+  @Override
+  @PreFetch(enabled = false)
+  public void onCustomizedViewRootChange(List<String> customizedViewTypes,
```

Comment:
So the prefetched type is not used anywhere. I think we should add listeners to the intersection of the available types and the user required types, right?

Reply:
I added prefetch to the code in case we want to enable it in the future. However, we do not use it here. At the end of the day, I think intersection does not matter and here we only need to register what user requested. Please let me know if that answers your question.